core/OrderAnalyser.py

```diff
@@ -40,11 +40,19 @@ def start(self):
 
         if not launched_synapses:
             Utils.print_info("No synapse match the captured order: %s" % self.order)
-        else:
-            for synapse in launched_synapses:
-                params = self._get_synapse_params(synapse, self.order)
-                for neuron in synapse.neurons:
-                    self._start_neuron(neuron, params)
+
+        if not launched_synapses and self.main_controller.settings.default_synapse is not None:
+            default_synapse = self._get_default_synapse(self.brain.synapses, self.main_controller.settings.default_synapse)
+
+            if default_synapse is not None:
+                logger.debug("Default synapse found %s" % default_synapse)
+                Utils.print_info("Default synapse found: %s, running it" % default_synapse.name)
+                launched_synapses.append(default_synapse)
+
+        for synapse in launched_synapses:
```

Comment:
If we don't enter the if, then launched_synapses is empty so it shouldn't loop, does it ? 
(no expert in python, so maybe it's true only in other language?)